File: javascript-solutions/maximum-units-on-a-truck.js
/**
 * @param {number[][]} boxTypes
 * @param {number} truckSize
 * @return {number}
 */
var maximumUnits = function (boxTypes, truckSize) {
	boxTypes.sort((a, b) => a[1] - b[1]);
	let maxUnits = 0;
	while (truckSize > 0 && boxTypes.length > 0) {
		const currentBox = boxTypes.pop();
		if (currentBox[0] <= truckSize) {
			maxUnits += currentBox[0] * currentBox[1];
		} else {
			maxUnits += truckSize * currentBox[1];
		}
		truckSize -= currentBox[0];
	}
	return maxUnits;
};

File: python-solutions/maximum-units-on-a-truck.py
from typing import List


class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(key=lambda x: x[1])
        maxUnits = 0
        while truckSize > 0 and len(boxTypes) > 0:
            currentBox = boxTypes.pop()
            if currentBox[0] <= truckSize:
                maxUnits += currentBox[0] * currentBox[1]
            else:
                maxUnits += truckSize * currentBox[1]
            truckSize -= currentBox[0]
        return maxUnits
